website/src/firebase-utils/login.jsx

```diff
@@ -0,0 +1,55 @@
+import * as firebase from 'firebase';
+
+/** 
+ * Opens login prompt for user and redirects them to the home page if successful.
+ */
+function promptForLogin() {
+  const provider = new firebase.auth.GoogleAuthProvider();
+  firebase.auth().signInWithPopup(provider).then(result => {
+    const user = result.user;
+    const uid = user.uid;
+    const user_email = user.email;
+    const user_displayName = user.displayName;
+    const user_emailVerified = user.emailVerified;
+    const user_isAnonymous = user.isAnonymous;
+    const user_providerData = user.providerData[0];
+    const user_providerId = user.providerId;
+
+    // check if account already exists, if not add an entry.
+    const accountCheck = firebase.database().ref('users/' + uid);
+    accountCheck.once("value", snapshot => {
+      if (snapshot.val() === null) {
+        accountCheck.set({
```

Comment:
i just realized that we will need to do this on the backend, since users shouldn't be able to modify everything on their user account (ex: whether they are an admin or not, their grid list, etc). We'll need to implement this in the backend and send a request.